pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -481,6 +481,11 @@ stdenv.mkDerivation {
       substituteAll ${../wrapper-common/utils.bash} $out/nix-support/utils.bash
     ''
 
+    + optionalString stdenv.targetPlatform.isDarwin ''
+      echo "-arch ${targetPlatform.darwinArch}" >> $out/nix-support/cc-cflags
+      echo "-m${targetPlatform.darwinPlatform}-version-min=${targetPlatform.darwinMinVersion}" >> $out/nix-support/cc-cflags-before
```

Comment:
Clang allows both `macos` and `macosx`:

* https://github.com/llvm/llvm-project/blob/llvmorg-11.1.0/clang/include/clang/Driver/Options.td#L2225-L2228

The former was introduced here:

* https://github.com/llvm/llvm-project/commit/921ab758548f2b9b611a4769c1d34ab46650ac5d

But GCC (working on a PR) only takes `macosx`:

* https://github.com/gcc-mirror/gcc/blob/releases/gcc-10.2.0/gcc/config/darwin.opt#L65-L69

This results in errors when compiling anything with GCC:

```
g++ -O2 -g rig.cc -o rig -Wall -DDATADIR="\"/nix/store/dlz0hkf17qdh84cnpjamawi0p41y8qcj-rig-1.11/share/rig\""
g++: error: unrecognized command-line option '-mmacos-version-min=11.0'; did you mean '-mmacosx-version-min='?
```

Should we switch back to `macosx` or do something else?

Reply:
Looking at the commit message during the introduction in llvm, and the code of `cctools`, it looks like `macos` is intended to be the canonical name for the platform, and `macosx` is a supported alias. So I'd like to keep `darwinPlatform` to the modern `macos` version, and add an exception for gcc in the cc-wrapper.